src/mscorlib/src/System/Environment.cs

```diff
@@ -1519,8 +1583,11 @@ public static string UserDomainName
         {
             [System.Security.SecuritySafeCritical]  // auto-generated
             get {
+#if PLATFORM_UNIX
+                return MachineName;
```

Comment:
is this the best we can do on unix? If there's no "redmond\danmose" I think I would expect the user name "danmose"